### ISSUE TITLE
Missing scale parameter causes a rounding problem during binding

### ISSUE THREAD

**Issue by mloskot** [nanodbc/nanodbc#87]
_From @dedomilo on September 24, 2015 0:33_

When I try to bind a double variable 12345.987 to a `DECIMAL(8,3)` column during insert in SQL Server, it ends up rounded to 12346 in the database table. Strangely enough the same code works fine with Oracle `NUMBER(8,3)` column.

It looks like the scale parameter returned by the `SQLDescribeParam` call (ignored in `prepare_bind` function) is not propagated to the following `SQLBindParameter` call (forced to 0 in `bind_parameter` functions and in `bind_null` function).

This is the fix which worked for me although you probably can come up with a better solution:
- Add a scale parameter to `prepare_bind` function and return in it the number of decimal digits reported by the ODBC driver:
  
  ```
  void prepare_bind(
      short param
      , std::size_t elements
      , param_direction direction
      , SQLSMALLINT& data_type
      , SQLSMALLINT& param_type
      , SQLULEN& parameter_size
      , SQLSMALLINT& scale)  // <---
  {
      RETCODE rc;
      NANODBC_CALL_RC(
          SQLDescribeParam
          , rc
          , stmt_
          , param + 1
          , &data_type
          , &parameter_size
          , &scale // <---
  ```
- Get that additional value from the 4 calls to `prepare_bind`:
  
  ```
  SQLSMALLINT scale;
  prepare_bind(... original parameters ..., scale);
  ```
- Add a scale parameter to the `bind_parameter` functions and use it in the `SQLBindParameter` calls:
  
  ```
  template<class T>
  void bind_parameter(
      short param
      , const T* data
      , std::size_t /*elements*/
      , SQLSMALLINT data_type
      , SQLSMALLINT param_type
      , SQLULEN parameter_size
      , SQLSMALLINT scale) // <---
  {
      RETCODE rc;
      NANODBC_CALL_RC(
          SQLBindParameter
          , rc
          , stmt_ // handle
          , param + 1 // parameter number
          , param_type // input or output type
          , sql_ctype<T>::value // value type
          , data_type // parameter type
          , parameter_size // column size ignored for many types, but needed for strings
          , scale // decimal digits   <---
  ```
- I did the same with the other `bind_parameter` function.
- Send the scale parameter in the 3 `bind_parameter` calls:
  
  ```
  bind_parameter(... original parameters ..., scale);
  ```

Most likely the original code worked with Oracle because in the `SQLDescribeParam` call the Oracle ODBC driver reported the `NUMBER(8,3)` column as SQL type 12 (SQL_VARCHAR) while SQL Server reported the `DECIMAL(8,3)` column as SQL type 3 (SQL_DECIMAL).

I hope you can fix this in one of the next versions.

I really like the library and I appreciate the work you've done.

_Copied from original issue: lexicalunit/nanodbc#49_

**Comment by mloskot** [comment]
_From @lexicalunit on September 24, 2015 1:28_

Very cool! If you'd like to fork my repository and commit your changes in a branch and submit a PR I'd be more than happy to merge it in. I would probably add a test case for `12345.987` too, if you haven't already. The method you described for passing on the scale value seems perfectly reasonable to me.

**Comment by mloskot** [comment]
_From @dedomilo on September 24, 2015 3:7_

Please do this change yourself. I barely understand the library’s code and I’m completely new to GitHub contributions. I had enough thrill today trying to figure out how the Markdown works so I could post a somewhat readable message.

**Comment by mloskot** [comment]
_From @lexicalunit on September 24, 2015 15:0_

No problem, I'll make sure to credit you in the commit message. Thanks again for the awesome issue documentation and fix!

**Comment by mloskot** [comment]
_From @lexicalunit on September 24, 2015 18:0_

This issue and issue #48 have been resolved in release [2.6.0](https://github.com/lexicalunit/nanodbc/releases/tag/v2.6.0).
